### ISSUE TITLE
Cannot open (some?) h5 files with current version of pytables (3.4.4)

### ISSUE THREAD

**Issue by David-Baddeley** [python-microscopy/python-microscopy#50]
**[Original report](https://bitbucket.org/david_baddeley/python-microscopy/issue/50) by Andrew Barentine (Bitbucket: [AndrewBarentine](https://bitbucket.org/AndrewBarentine), ).**

----------------------------------------

Bug reported by others on windows with conda install, recreated by me with the current version of our repo on osx. 

I could open their h5 file, until I performed the following update:

```
#!python

    ca-certificates: 2017.08.26-ha1e5d58_0 --> 2018.03.07-0
    certifi:         2018.1.18-py27_0      --> 2018.4.16-py27_0
    conda:           4.4.11-py27_0         --> 4.5.4-py27_0
    hdf5:            1.10.1-h6090a45_0     --> 1.10.2-hfa1e0ec_1
    openssl:         1.0.2n-hdbc3d79_0     --> 1.0.2o-h26aff7b_0
    pytables:        3.4.2-py27ha4551b8_2  --> 3.4.4-py27h5ca999c_0
```

After this update, opening the file results in the following traceback:

```
#!python

$ dh5view ~/Downloads/censored_filename.h5
Starting main loop
Loading data
filename == /Users/Andrew/Downloads/censored_filename.h5
Traceback (most recent call last):
  File "/Users/Andrew/anaconda/lib/python2.7/site-packages/wx-3.0-osx_cocoa/wx/_core.py", line 16766, in <lambda>
    lambda event: event.callable(*event.args, **event.kw) )
  File "/Users/Andrew/python-microscopy/PYME/DSView/dsviewer.py", line 414, in LoadData
    im = ImageStack(filename=args[0], queueURI=options.queueURI, mdh=options.metadata)
  File "/Users/Andrew/python-microscopy/PYME/IO/image.py", line 221, in __init__
    self.Load(filename, prompt=load_prompt)
  File "/Users/Andrew/python-microscopy/PYME/IO/image.py", line 945, in Load
    self._loadh5(filename)
  File "/Users/Andrew/python-microscopy/PYME/IO/image.py", line 424, in _loadh5
    self.dataSource = HDFDataSource.DataSource(filename, None)
  File "/Users/Andrew/python-microscopy/PYME/IO/DataSources/HDFDataSource.py", line 39, in __init__
    if 'PZFImageData' in dir(self.h5File.root):
  File "/Users/Andrew/anaconda/lib/python2.7/site-packages/tables/group.py", line 828, in __dir__
    subnods = [c for c in self._v_children if c.isidentifier()]
AttributeError: 'str' object has no attribute 'isidentifier'
Closing remaining open files:/Users/Andrew/Downloads/censored_filename.h5...done
```

I can no longer open my (basic) h5 files either after the update.

**Comment by David-Baddeley** [comment]
**Original comment by Andrew Barentine (Bitbucket: [AndrewBarentine](https://bitbucket.org/AndrewBarentine), ).**

----------------------------------------

Note that reverting, i.e. 
```
#!python

    hdf5:     1.10.2-hfa1e0ec_1    --> 1.10.1-ha036c08_1
    pytables: 3.4.4-py27h5ca999c_0 --> 3.4.2-py27ha4551b8_2
```

again allows me to open the files

**Comment by David-Baddeley** [comment]
**Original comment by David Baddeley (Bitbucket: [david_baddeley](https://bitbucket.org/david_baddeley), ).**

----------------------------------------

OK - I'll pin the pytables version in meta.yaml

**Comment by David-Baddeley** [comment]
**Original comment by Andrew Barentine (Bitbucket: [AndrewBarentine](https://bitbucket.org/AndrewBarentine), ).**

----------------------------------------

Should be resolved with commit ac679b4

**Comment by David-Baddeley** [comment]
**Original comment by Zach Connerty-Marin (Bitbucket: [zacsimile](https://bitbucket.org/zacsimile), GitHub: [zacsimile](https://github.com/zacsimile)).**

----------------------------------------

This is still an issue on Windows. Did an install with the latest version of PYME this morning. Received the following error.


```
#!python

Creating Pipeline
Traceback (most recent call last):
  File "c:\users\zrc4\documents\python-microscopy\PYME\LMVis\visCore.py", line 377, in OnOpenFile
    self.OpenFile(filename)
  File "c:\users\zrc4\documents\python-microscopy\PYME\LMVis\visCore.py", line 665, in OpenFile
    self.pipeline.OpenFile(filename, **args)
  File "c:\users\zrc4\documents\python-microscopy\PYME\LMVis\pipeline.py", line 654, in OpenFile
    ds = self._ds_from_file(filename, **kwargs)
  File "c:\users\zrc4\documents\python-microscopy\PYME\LMVis\pipeline.py", line 560, in _ds_from_file
    ds = tabular.h5rDSource(filename)
  File "c:\users\zrc4\documents\python-microscopy\PYME\IO\tabular.py", line 277, in __init__
    h5rSource.__init__(self, h5fFile, 'DriftResults')
  File "c:\users\zrc4\documents\python-microscopy\PYME\IO\tabular.py", line 217, in __init__
    if not tablename in dir(self.h5f.root):
  File "C:\Users\zrc4\AppData\Local\conda\conda\envs\pyme_test\lib\site-packages\tables\group.py", line 828, in __dir__
    subnods = [c for c in self._v_children if c.isidentifier()]
AttributeError: 'str' object has no attribute 'isidentifier'
```

This was fixed by running conda install pytables=3.2.2.

**Comment by David-Baddeley** [comment]
**Original comment by Andrew Barentine (Bitbucket: [AndrewBarentine](https://bitbucket.org/AndrewBarentine), ).**

----------------------------------------

Just checked - @zacsimile ran into this problem because the pyme-depends package on anaconda hasn't been updated. Could you take care of that, @david_baddeley?
